### ISSUE TITLE
Incorrect error messages from Auth.signUp

### ISSUE THREAD

**Issue by kimfucious** [aws-amplify/amplify-js#5532]
**Describe the bug**

When running the following command with a known bad property for `phone_number`:

```js
const cognitoUser = await Auth.signUp({
  username: values.username,
  password: values.password,
  attributes: {
    email: values.email,
    phone: values.phone, <= bad property (should be phone_number)
  }
});
```
The following error is logged to the browser console and in Hub when the auth listener "hears" `signUp_failure`.
```
"Username could not be created. Please make sure that the username you have entered is between 1 and 128 characters."
```
🤔 I *knew* that the username was there, but I did wind up chasing this red herring for longer than I'd like to admit.

Finally, I checked the HTTP response in the dev-tools network and found:

```console
{"__type":"InvalidParameterException","message":"Attributes did not conform to the schema: phone: Attribute does not exist in the schema.\n"}
```

**To Reproduce**

1. Use a bad property when calling `Auth.signUp`
2. View error in console
3. Compare with HTTP response in network tab
4. See difference

**Expected behavior**
The error in the console should be applicable for the actual error

This discrepancy most likely occurs for errors other than an incorrect property as described above.

**Comment by amhinson** [comment]
Similar to: https://github.com/aws-amplify/amplify-js/issues/5313

Looks like a fix was pushed for this recently. Can you try using `aws-amplify@unstable`?

**Comment by kimfucious** [comment]
Hi @amhinson, thanks for the follow up.

Yes, #5313 is pretty much what I'm seeing.

I dread using anything with the word unstable in it, and it's really not a blocker at this point.

So I'm fine sticking with stable for the now, knowing that you folks are already aware of this issue.

I'll leave this open for you to close, tag, or process per your workflow.

**Comment by amhinson** [comment]
We just released a new version (3.0.9) that should address this issue (among other things). Give it a go and let us know if anything else comes up 👍
